src/app/dashboard/dashboard.component.ts

```diff
@@ -1,23 +1,106 @@
-import { Component, OnInit } from '@angular/core';
-import { Hero } from '../hero';
-import { HeroService } from '../hero.service';
+import { takeUntil } from 'rxjs/operators';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+import { Weapon } from '../core/interface/weapon';
+import { cloneData } from '../core/utils/utils';
+import { Hero } from '../core/interface/hero';
+import { WeaponService } from '../weapon/weapon.service';
+import { GameService } from './game.service';
+import { MessageService } from '../message.service';
+import { HeroService } from '../heroes/hero.service';
 
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: [ './dashboard.component.css' ]
 })
-export class DashboardComponent implements OnInit {
+export class DashboardComponent implements OnInit, OnDestroy {
   heroes: Hero[] = [];
+  weapons: Weapon[] = [];
+  heroesPlaying: Hero[] = [];
+  $unSubscribe = new Subject();
 
-  constructor(private heroService: HeroService) { }
+  constructor(
+    private readonly heroService: HeroService,
+    private readonly weaponService: WeaponService,
+    private readonly gameService: GameService,
+    private readonly messageService: MessageService
+  ) { }
 
   ngOnInit() {
+    this.heroesPlaying = this.gameService.heroesPlaying.value || [];
     this.getHeroes();
+    this.getWeapons();
+    this.gameService.heroesPlaying
+    .pipe(takeUntil(this.$unSubscribe))
+    .subscribe((heroes: Hero[]) => {
+      this.heroesPlaying = heroes;
+    });
   }
 
+  ngOnDestroy(): void {
+    this.$unSubscribe.next();
+  }
+
+  /**
+   * get heroes
+   */
   getHeroes(): void {
     this.heroService.getHeroes()
-      .subscribe(heroes => this.heroes = heroes.slice(1, 5));
+      .subscribe(heroes => this.heroes = heroes);
+  }
+
+  /**
+   * get weapons
+   */
+  getWeapons(): void {
+    this.weaponService.getWeapons()
+      .subscribe(weapons => this.weapons = weapons);
+  }
+
+  /**
+   * check hero active
+   * @param id: hero id
+   * @returns: is active
+   */
+  isActive(id: number): boolean {
+    return !!this.heroesPlaying.find(hero => hero.id === id);
+  }
+
+  /**
+   * Toggle hero playing
+   * @param status: is playing
+   * @param id: hero id
+   */
+  toggleHero(status: boolean, hero: Hero): void {
+    if (status) {
+      const heroClone: Hero = cloneData(hero);
+      heroClone.playingHealth =  heroClone.health + heroClone.armour?.health;
+      this.heroesPlaying.push(heroClone);
+      this.messageService.add(`Dashboard: Adding hero ${heroClone.name}`);
+    } else {
```

Comment:
looks like we broke the Observable stream here again by cloning the hero instead of adding the observable to heroesPlaying.  Now if we change the heroes name or weapong the game player will not be aware.  Well we could write a whole bunch of other unncessary code instead to keep it aware but that defeats the purpose of Observables.

Reply:
My idea is heroes can join the battle and hero's health in the battle don't impact to hero's health in the hero card. So I do the heroClone for heroesPlaying. But it seems you want to add update armour above, I will update this logic, and commit it later